### PR TITLE
New rule: Sorted

### PR DESCRIPTION
**Pull request by Reeywhaar**
Handles #835 
I also made docs md file, which contibuting.md doesn't mention. Hope it helps. But changelog there must be edited.
Thanks!

### CODE REVIEW

**Comment by nickl-** [comment]
The use of the $cmp variable, albeit concise, makes the code less readable at quick glance. 

Perhaps something like 
```PHP
if (($this->ascending && ($this->fn)($input[$i]) < ($this->fn)($input[$i - 1])) 
   || (!$this->ascending && ($this->fn)($input[$i]) > ($this->fn)($input[$i - 1])))
return false;

```
would be more legible without sacrificing too much verbosity while eliminating the need for a local variable.

Otherwise good work, I like it  +1!

**Comment by nickl-** [comment]
Wow quick response, good job!

At the risk of being pedantic, some code style fixes pending.

Spaces between statements and `(` brackets. To distinguish statements (`if for while`) from functions/objects.
Spaces between `)` brackets and `{` braces.
Single line `if` statements don't require begin and end. 

Won't stop this because of the new line after opening the if statement but that doesn't seem necessary. I would rather add a new line and indentation before the return at `if($count < 2) return true;` as a convention. 

My bad for not spotting this first time round.

**Comment by Reeywhaar** [comment]
Thanks @nickl-! It's great. Codestyle is important. Hope I got it right.

**Comment by nickl-** [comment]
Please see code review comments.

**Comment by Reeywhaar** [comment]
Oh, yes I saw it. Something wrong with the commit?

```php
if ($count < 2) {
    return true;
}
```
Here I decided to make brackets, because I'd rather avoid multiline if without brackets.

```php
if (
    ($this->ascending && ($this->fn)($input[$i]) < ($this->fn)($input[$i - 1]))
    || (!$this->ascending && ($this->fn)($input[$i]) > ($this->fn)($input[$i - 1]))
) {
```
Does this newlines bother you?
If you could write in code what do you mean, it will help me to understand a lot.

**Comment by nickl-** [comment]
Not sure what happened to this post but here goes again.

> Oh, yes I saw it. Something wrong with the commit?
>
> if ($count < 2) {
>    return true;
> }

We don't add the braces for single line statements.

ex.
```PHP
if ($count < 2) 
    return true;
```

> Here I decided to make brackets, because I'd rather avoid multiline if without brackets.
> 
> if (
>    ($this->ascending && ($this->fn)($input[$i]) < ($this->fn)($input[$i - 1]))
>    || (!$this->ascending && ($this->fn)($input[$i]) > ($this->fn)($input[$i - 1]))
> ) {
>
> Does this newlines bother you?
> If you could write in code what do you mean, it will help me to understand a lot.

Would prefer:

```PHP
if (($this->ascending && ($this->fn)($input[$i]) < ($this->fn)($input[$i - 1]))
   || (!$this->ascending && ($this->fn)($input[$i]) > ($this->fn)($input[$i - 1]))) 
     return false:
```

But will allow:
```PHP
if (
   ($this->ascending && ($this->fn)($input[$i]) < ($this->fn)($input[$i - 1]))
   || (!$this->ascending && ($this->fn)($input[$i]) > ($this->fn)($input[$i - 1]))
) 
  return false:
```
Which probably looks better...

**Comment by nickl-** [comment]
Perfect! Good job! 

Waiting for travis to complete...

**Comment by nickl-** [comment]
Merged.

Looking forward to working with you again...

**Comment by Reeywhaar** [comment]
@nickl- Thanks for guiding me!

**Comment by nickl-** [comment]
You are welcome! 

Why don't you try and implement one of the open [New Validator](https://github.com/Respect/Validation/issues?q=is%3Aissue+is%3Aopen+label%3A%22New+Validator%22) requests and I'll assist you some more.

File: docs/Sorted.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        'key' => 4,
+    ],
+]); // false
+```
+
+## Changelog
+
+Version | Description
+--------|-------------
+  1.1.1 | Created
```
Comment (henriquemoody) [inline docs/Sorted.md]: That would be 2.0.0

**Comment by nickl-** [comment]
When we planning on releasing 2.0? 

I'll make a reminder to fix this before then.

**Comment by henriquemoody** [comment]
I'm counting on when we can finish #678... But I'm not even sure anymore about it

**Comment by nickl-** [comment]
Ok so we can update this as part of #678.

**Pull request merged**